utility_network/display-utility-associations/src/main/java/com/esri/samples/display_utility_associations/DisplayUtilityAssociationsSample.java

```diff
@@ -0,0 +1,244 @@
+      controlsVBox.setMaxSize(100, 50);
+      controlsVBox.setAlignment(Pos.BASELINE_LEFT);
+      controlsVBox.getStyleClass().add("panel-region");
+      controlsVBox.getChildren().addAll(attachmentLabel, attachmentImageView, connectivityLabel, connectivityImageView);
+      
+      // add the scene view and controls to the stack pane
+      stackPane.getChildren().addAll(mapView, controlsVBox);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+      
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+    
+  }
+  
+  /**
+   * Get utility association type for each association within extent , create graphics and add it to the graphics overlay.
+   */
+  private void addAssociationsGraphics() {
+    try {
+      // check if the current viewpoint is outside of the max scale
+      if (mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE).getTargetScale() >= 2000)
+        return;
+      
+      // check if the current viewpoint has an extent
+      Envelope extent = mapView.getCurrentViewpoint(Viewpoint.Type.BOUNDING_GEOMETRY).getTargetGeometry().getExtent();
+      if (extent != null) {
+        // get all of the associations in the extent of the viewpoint
+        ListenableFuture<List<UtilityAssociation>> associationsFuture = utilityNetwork.getAssociationsAsync(extent);
+  
+        associationsFuture.addDoneListener(() -> {
+    
+          try {
+            List<UtilityAssociation> associations = associationsFuture.get();
+      
+            associations.forEach(association -> {
+              // check if the graphics overlay already contains the association
+              if (associationsOverlay.getGraphics().stream().noneMatch(graphic ->
+                graphic.getAttributes().containsKey("GlobalId")
+                  && graphic.getAttributes().get("GlobalId") == association.getGlobalId())
+                && association.getGeometry() != null) {
+          
+                // create attributes for the graphic so that the renderers display the appropriate symbol
+                Map<String, Object> attributes = Map.of("GlobalId", association.getGlobalId(), "AssociationType", association.getAssociationType().toString());
+                // create and add the graphic for the association
+                Graphic newGraphic = new Graphic(association.getGeometry(), attributes);
+                associationsOverlay.getGraphics().add(newGraphic);
+              }
+            });
+          } catch (InterruptedException | ExecutionException e) {
+            new Alert(Alert.AlertType.ERROR, "Error " + e.getMessage()).show();
+          }
+    
+        });
+      }
+    }
+    // this is thrown when there are too many associations in the extent
+    catch (Exception e) {
+      new Alert(Alert.AlertType.ERROR, "too many associations" + e.getMessage()).show();
+    }
+  }
```

Comment:
This is great - I can only suggest a little bit of formatting changes here so that it's a little easier to read (especially the stream):
```suggestion
    try {
      // check if the current viewpoint is outside of the max scale
      if (mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE).getTargetScale() >= 2000)
        return;

      // check if the current viewpoint has an extent
      Envelope extent = mapView.getCurrentViewpoint(Viewpoint.Type.BOUNDING_GEOMETRY).getTargetGeometry().getExtent();
      if (extent != null) {

        // get all of the associations in the extent of the viewpoint
        ListenableFuture<List<UtilityAssociation>> associationsFuture = utilityNetwork.getAssociationsAsync(extent);
        associationsFuture.addDoneListener(() -> {
          try {
            List<UtilityAssociation> associations = associationsFuture.get();

			// iterate through all associations
            associations.forEach(association -> {

              // check if the graphics overlay already contains the association
              if (association.getGeometry() != null &&
                      associationsOverlay.getGraphics().stream().noneMatch(graphic ->
                              graphic.getAttributes().containsKey("GlobalId") &&
                                      graphic.getAttributes().get("GlobalId") == association.getGlobalId())) {

                // create attributes for the graphic so that the renderers display the appropriate symbol
                Map<String, Object> attributes = Map.of(
                        "GlobalId", association.getGlobalId(),
                        "AssociationType", association.getAssociationType().toString());

                // create and add the graphic for the association
                Graphic newGraphic = new Graphic(association.getGeometry(), attributes);
                associationsOverlay.getGraphics().add(newGraphic);
              }

            });
          } catch (InterruptedException | ExecutionException e) {
            new Alert(Alert.AlertType.ERROR, "Error getting the utility associations").show();
          }

        });
      }
    }
    // this is thrown when there are too many associations in the extent
    catch (Exception e) {
      new Alert(Alert.AlertType.ERROR, "Too many associations found").show();
    }
  }
```